js/restaurant.js

```diff
@@ -414,6 +431,22 @@ function fireRule(rule) {
   }
 
   if(win){
+    if (progress.guessHistory[currentLevel]) {
+      if (progress.guessHistory[currentLevel].solutions) {
+        if (!progress.guessHistory[currentLevel].solutions.includes(rule)) {
+          $(".display-help .solutions-title").show();
+          let solution = $("<solution>" + rule + "</solution>");
+          $(".display-help .solutions").append(solution);
+        }
+      }
+    }
+    else {
+      console.log('shit')
+      $(".display-help .solutions-title").show();
+      let solution = $("<solution>" + rule + "</solution>");
+      $(".display-help .solutions").append(solution);
+    }
+
```

Comment:
If there are `solutions` check the input `rule` if it matches existing `solution`. If it's new then add it to `solutions container`.
If there were no `solutions` previously you need to show `solutions-title` also